Runtime/System/ClientSimSceneManager.cs

```diff
@@ -114,6 +114,22 @@ public Transform GetSpawnPoint(bool remote = false)
             // Fallback to first spawn point
             return _descriptor.spawns[0];
         }
+
+        public Transform GetSpawnPoint(int index)
+        {
+            if (!HasSceneDescriptor())
+            {
+                throw new ClientSimException("Cannot get spawn point when there is no scene descriptor.");
+            }
+            
+            if (_descriptor.spawns.Length <= index || _descriptor.spawns[index] == null)
```

Comment:
Can you also check if index is greater than or equal to zero? If someone puts a negative number, the error will be an array out of bounds exception rather than a ClientSimException.

Reply:
Yes - added this check to revert to 0 if the index is invalid.